pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -104,9 +109,10 @@ func (srv PrometheusSrv) RouteGetRuleStatuses(c *models.ReqContext) response.Res
 	}
 
 	alertRuleQuery := ngmodels.ListAlertRulesQuery{
-		OrgID:        c.SignedInUser.OrgId,
-		DashboardUID: dashboardUID,
-		PanelID:      panelID,
+		OrgID:         c.SignedInUser.OrgId,
+		DashboardUID:  dashboardUID,
+		PanelID:       panelID,
+		NamespaceUIDs: namespaceUIDs,
```

Comment:
You will need to add `namespace_uid` to the `IDX_alert_rule_org_id_dashboard_uid_panel_id` index to avoid full table scan https://github.com/grafana/grafana/pull/39660#issuecomment-929035323.

Reply:
I'm going to remove this from the current PR and we can chat about it separately. Good shout on the full table scan.